### PR TITLE
Add duplicate ID check error, hintHandler test revamp

### PR DESCRIPTION
**Pull request by Anyhowclick**

### CODE REVIEW

File: contractsSol5/KyberHintHandler.sol [hunk 1]
```diff
@@ -256,8 +256,13 @@ contract KyberHintHandler is IKyberHint, Utils4 {
             if (reserveIds.length != splits.length) return false;
 
             uint bpsSoFar;
+            bytes8[] memory checkDuplicateIds = new bytes8[](reserveIds.length);
             for (uint i = 0; i < splits.length; i++) {
                 bpsSoFar += splits[i];
+                for (uint j = 0; j < checkDuplicateIds.length; j++) {
```
Comment (ilanDoron) [inline contractsSol5/KyberHintHandler.sol]: nice
Now this function returns error codes
so should update this.

File: test/sol5/kyberMatchingEngine.js [hunk 2]
```diff
@@ -1663,6 +1663,56 @@ contract('KyberMatchingEngine', function(accounts) {
                 );
             });
 
+            describe(`should return zero rates and amounts if duplicate reserve ID is used for split`, async() => {
```
Comment (ilanDoron) [inline test/sol5/kyberMatchingEngine.js]: should add hint handler test as well.

File: contractsSol5/KyberHintHandler.sol (hunk 1 above)
Comment (Anyhowclick) [reply contractsSol5/KyberHintHandler.sol]: updated

File: test/sol5/kyberMatchingEngine.js (hunk 2 above)
Comment (Anyhowclick) [reply test/sol5/kyberMatchingEngine.js]: added

**Comment by Anyhowclick** [comment]
- Use expectRevert instead of try-catch
- Move some "it" tests to before block so that .only works

File: test/sol5/kyberMatchingEngine.js (hunk 2 above)
Comment (Anyhowclick) [reply test/sol5/kyberMatchingEngine.js]: Done. Also, helped to "re-write" the tests, cos I started fixing 1 small thing that led to fixing another small thing, that led to fixing a big thing.

Just like this: https://www.youtube.com/watch?v=AbSehcT19u0

**Pull request merged**